src/render_engine/vulkan/vulkan_render_engine.cpp

```diff
@@ -505,42 +505,55 @@ namespace nova::renderer::rhi {
             VkWriteDescriptorSet vk_write = {};
             vk_write.sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
             vk_write.dstSet = static_cast<const VulkanDescriptorSet*>(write.set)->descriptor_set;
-            vk_write.dstBinding = write.binding;
-            vk_write.descriptorCount = 1;
+            vk_write.dstBinding = write.first_binding;
+            vk_write.descriptorCount = static_cast<uint32_t>(write.bindings.size());
             vk_write.dstArrayElement = 0;
 
             switch(write.type) {
                 case DescriptorType::CombinedImageSampler: {
-                    VkDescriptorImageInfo vk_image_info = {};
-                    vk_image_info.imageView = image_view_for_image(write.image_info.image);
-                    vk_image_info.imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
-                    vk_image_info.sampler = static_cast<VulkanSampler*>(write.image_info.sampler)->sampler;
-
-                    image_infos.push_back(vk_image_info);
-
+                    const auto write_begin_idx = image_infos.size();
+
+                    std::transform(write.bindings.begin(),
```

Comment:
This is a long philosophical debate, but I feel like this is much better as a standard ranged for.